packages/api/src/middleware/auth.ts

```diff
@@ -106,8 +108,14 @@ function authFactory(params: AuthParams): RequestHandler {
     }
     const accessRules = tokenObject?.access?.rules;
     if (accessRules) {
-      const fullPath = pathJoin2(req.baseUrl, req.path);
-      const { httpPrefix } = req.config;
+      let fullPath = pathJoin2(req.baseUrl, req.path);
+      let { httpPrefix } = req.config;
+      if (params.originalUriHeader) {
+        const header = req.headers[params.originalUriHeader];
+        const originalUri = new URL(header?.toString() ?? "");
```

Comment:
If `header` is empty this will throw error and return 500. 
Can we at least return 400?